client/components/happychat/button.jsx

```diff
@@ -17,11 +17,8 @@ import classnames from 'classnames';
  * Internal dependencies
  */
 import viewport from 'lib/viewport';
-import {
-	hasUnreadMessages,
-	isHappychatAvailable,
-	hasActiveHappychatSession,
-} from 'state/happychat/selectors';
+import { hasUnreadMessages, isHappychatAvailable } from 'state/happychat/selectors';
+import hasActiveHappychatSession from 'state/happychat/selectors/has-active-happychat-session';
```

Comment:
Do we really need to have a different file for each selector? I guess it adds to the build time and also more imports when composing selectors. Is this a new pattern in calypso?

Reply:
Yeah, it's a new Calypso pattern. Some of these selectors (the ones related to dotcom) should be moved to the `state/selectors` folder in upcoming PRs; others should stay here. At the moment is useful for me to have them all together so it's easier to understand the Happychat boundaries.